doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+<https://wiki.debian.org/DebianScience/LinearAlgebraLibraries>`__ and avoid
+documenting in more detail how this works on other distros.
+
+Note that Fedora is an exception; it is the only distro that ships FlexiBLAS
+(see the next section for more on that) and allows installing multiple BLAS
+libraries in parallel so true runtime switching without having to invoke the
+system package manager becomes possible. See `the Fedora docs on system-level
+and user-level selection
+<https://docs.fedoraproject.org/en-US/packaging-guidelines/BLAS_LAPACK/#_backend_selection>`__
+for more details.
+
+
+FlexiBLAS
+`````````
+
+`FlexiBLAS <https://github.com/mpimd-csc/flexiblas>`__ provides runtime
+switching support (among other things) for all installed BLAS libraries that it
+can detect. There are a few limitations at the time of writing (March 2024),
+primarily: no support for Windows, no support for macOS Accelerate (the updated
+version, with ``NEWLAPACK`` symbols). If those limitations don't matter for
+you, FlexiBLAS can be a quite useful tool for efficient debugging, including
+for versions of OpenBlAS and other BLAS libraries that you have to build from
+source.
+
+Once you have everything set up, the development experience is::
+
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
+    $ FLEXIBLAS=NETLIB python dev.py test -s linalg
+    $ FLEXIBLAS=OpenBLAS python dev.py test -s linalg
+    # Or export the environment variable to make the selection stick:
+    $ export FLEXIBLAS=OpenBLAS
+
+You can also provide a path to a built BLAS library (e.g.,
+``FLEXIBLAS="libbhlas_atlas.so"``) - see the `usage docs in its README
+<https://github.com/mpimd-csc/flexiblas#selecting-the-backend-at-runtime>`__
+for more details.
+
+Unless you're on Fedora, you will likely have to build FlexiBLAS from source,
+which is a bit of work. The good news is that this should work no matter if
+you're on Linux or macOS, and use Python via virtualenvs, conda environments,
+or in some other way. We'll go through how to build OpenBLAS and FlexiBLAS
+from source, to allow debugging whether something in the latest OpenBLAS
+version is different from Netlib BLAS/LAPACK (or MKL) or not.
+
+The below should work in any environment where you can build SciPy itself; the
+only additional tool we need is CMake (install with, for example, ``pip install
+cmake``).
+
+Clone each repository::
+
+    $ cd ..  # starting from the root of the local `scipy` repo
+    $ mkdir flexiblas-setup && cd flexiblas-setup
+    $ git clone https://github.com/OpenMathLib/OpenBLAS.git openblas
+    $ git clone https://github.com/mpimd-csc/flexiblas.git
+    $ mkdir built-libs  # our local prefix to install everything to
+
+Build OpenBLAS::
+
+    $ cd openblas
+    $ mkdir build && cd build
+    $ cmake .. -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
```

Comment:
The use of $PWD is a bit brittle here. A single extra Tab press adds a backslash and then things go to '$PWD' folder instead of ../../built-libs. Not sure if there's anything to do about it though.

Reply:
Nothing to be done about that I think; Ctrl-C of the instructions will work.